plugins/python/redisgears_python.c

```diff
@@ -1995,24 +1999,27 @@ static PyObject* DAGAddModelRun(PyObject *self, PyObject *args, PyObject *kargs)
         PyErr_SetString(GearsError, "Model name argument must be a string");
         return NULL;
     }
+    PyObject *ret = NULL;
     RAI_Error *err;
     RedisAI_InitError(&err);
     const char* modelNameStr = PyUnicode_AsUTF8AndSize(modelName, NULL);
 
     // Create MODELRUN op after bringing model from keyspace (raise an exception if it does not exist)
     RAI_DAGRunOp *modelRunOp = _createModelRunOp(modelNameStr, err);
     if (modelRunOp == NULL) {
-        RedisAI_FreeError(err);
-        return NULL;
+        goto cleanup;
```

Comment:
This is potentially a big bug, you initialize the inputsIter, outputsIter, modelOutputs. So if you reach the cleanup code from here you there will be junk inside them (and not NULL) and you will try to free junk.